### ISSUE TITLE
Compilation error

### ISSUE THREAD

**Issue by adamtrilling** [HashNuke/hound#115]
I'm getting a compiler error when trying to build hound:

```
==> hound
Compiling 32 files (.ex)

== Compilation error on file lib/hound/response_parsers/phantom_js.ex ==
** (ArgumentError) argument error
    (stdlib) erl_anno.erl:318: :erl_anno.set(:file, 'nofile', -1)
    (stdlib) erl_parse.yrl:1516: anonymous fn/3 in :erl_parse.map_anno/2
    (stdlib) erl_parse.yrl:1631: :erl_parse.modify_anno1/3
    (stdlib) erl_parse.yrl:1517: :erl_parse.map_anno/2
    (stdlib) erl_lint.erl:703: :erl_lint."-set_file/2-lc$^0/1-0-"/2
    (stdlib) erl_lint.erl:448: :erl_lint.exprs_opt/3
    (stdlib) erl_eval.erl:173: :erl_eval.check_command/2

could not compile dependency :hound, "mix compile" failed. You can recompile this dependency with "mix deps.compile hound", update it with "mix deps.update hound" or clean it with "mix deps.clean hound"
```

The error occurs on all three files in `lib/hound/response_parsers`, I guess depending on which order the compiler decides to compile them. I don't know enough about erlang to look too deeply into this.

I'm running elixir 1.3.1 and erlang 19.0. This is within a Phoenix 1.2 project, if that matters.

**Comment by tuvistavie** [comment]
@adamtrilling Hi, thanks for reporting.
I managed to reproduce the issue. It seems to occur when using Erlang 19.0. I am going to investigate.

**Comment by tuvistavie** [comment]
There seem to be an issue with the [`ResponseParser` `before_compile` hook](https://github.com/HashNuke/hound/blob/master/lib/hound/response_parser.ex#L16), no idea why.
I will investigate further.
@HashNuke Any idea?

**Comment by HashNuke** [comment]
@tuvistavie Installing 19.0 to investigate soon. Not sure why this happens.

**Comment by tuvistavie** [comment]
Ok, I just found out `quote line: -1` does not seem to work anymore,
replacing by `line: 0` gives the same result and removes the compile error.
I will update the code here and report to Elixir just to be sure.

**Comment by tuvistavie** [comment]
After checking the behavior, I was thinking that `line: -1` was needed for the
function to be evaluated as the last clause, but it seems that I was wrong, so
I will just remove the `line` option.

**Comment by adamtrilling** [comment]
And now it compiles!

Thanks!

**Comment by tuvistavie** [comment]
I just found out that the `line` option was needed for OTP 17.x, so I will revert to `line: 0`.
I will publish a new version as soon as Travis tells me we're good :smile:

**Comment by tuvistavie** [comment]
I just published 1.0.1 with support for OTP 19.
